checker.cpp

```diff
@@ -1,17 +1,228 @@
-#include <assert.h>
-
-bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
-    return false;
-  } else if(spo2 < 90) {
-    return false;
-  } else if(respRate < 30 || respRate > 95) {
-    return false;
-  }
-  return true;
+#include<assert.h>
+#include<iostream>
+#include<map>
+#include <vector>
+#include<iterator>
+#include <string>
+using namespace std;
+
+struct vitalsinfo {
+	string name;
+	float upperlimit;
+	float lowerlimit;
+	float singlelimit;
+
+	vitalsinfo(string vtname, float upper, float lower, float single) {
+		this->name = vtname;
+		this->upperlimit = upper;
+		this->lowerlimit = lower;
+		this->singlelimit = single;
+	}
+
+}; vector<vitalsinfo> vitaldetails;
+
+void addVitalDetails() {
+
+	struct vitalsinfo bpmobj("Bpm", 150, 70, 0);
+	struct vitalsinfo spo2obj("Spo2", 0, 0, 90);
+	struct vitalsinfo respobj("Resp", 95, 30, 0);
+
+	vitaldetails.push_back(bpmobj);
+	vitaldetails.push_back(spo2obj);
+	vitaldetails.push_back(respobj);
 }
 
+int getVitalIndex(string name) {
+	for(size_t index = 0; index < vitaldetails.size(); index++){
+		if (vitaldetails[index].name == name)
+			return index;
+	}
+}
+
+class Alert {
+public:
+	virtual void getAlertMessage(const std::map<string, int>& printresults) = 0;
+};
+
+class AlertInSMS :public Alert {
+public:
+	void getAlertMessage(const std::map<string, int>& printresults) {
+		std::map<string, int>::const_iterator itr;
+		for (itr = printresults.begin(); itr != printresults.end(); itr++) {
+			if (itr->second == 1)
+				cout << "Alert in SMS: " << itr->first << endl;
+		}
+	}
+};
+
+class AlertInSound :public Alert {
+public:
+	void getAlertMessage(const std::map<string, int>& printresults) {
+		std::map<string, int>::const_iterator itr;
+		for (itr = printresults.begin(); itr != printresults.end(); itr++) {
+			if (itr->second == 1)
+				cout << "Alert in Sound: " << itr->first << endl;
+		}
+	}
+};
+
+class Bpm {
```

Comment:
This class is very similar to Resp and Spo2. This code will need to be copied every time a vital is added. Can you make a class like RangeChecker and send the name in its constructor?

Reply:
Sure, I will Sudeep, thanks!!

Reply:
I have removed the redundancy, Sudeep Thanks!!